test/ocsp/helper/helper.go

```diff
@@ -181,10 +224,10 @@ func sendHTTPRequest(req []byte, ocspURL *url.URL) (*http.Response, error) {
 	return client.Do(httpRequest)
 }
 
-func getOCSPURL(cert *x509.Certificate) (*url.URL, error) {
+func getOCSPURL(cert *x509.Certificate, config CheckOCSPConfig) (*url.URL, error) {
 	var ocspServer string
-	if *urlOverride != "" {
-		ocspServer = *urlOverride
+	if config.urlOverride != "" {
+		ocspServer = config.urlOverride
```

Comment:
I realize this follows the original definition of the function, but it seems a bit perfunctory to pass in a config struct into this function, where it's only going to read `urlOverride` and then actually short-circuit the main "point" of the function if it's set. In terms of making the code more obvious I think it'd be a better tradeoff for the single call-site of this to check the config override and then skip calling `getOCSPURL` entirely. Presumably the URL in the config would have already been parsed and validated into a `url.URL` beforehand too, which moves the error-checking when the flag is specified closer to the interface of the program.

It's not a big deal since this is just an internal API and other may disagree on the effect of readability of this change, but if this were a public interface of some sort this pattern tends to increase coupling between parts and make testing and future changes more difficult.

Reply:
Yeah, this makes sense. I've reduced the API surface of the helper functions to take only the config parameters they actually need.

I haven't extracted url override to the point of not calling the helper function at all, because the helper does parse the url and return a Url object, rather than just a string. I'd leave other cleanup along those lines (e.g. making the urlOverride config value of type url.Url in the first place, so parsing+validation happens earlier) to a different change.